app/controllers/evaluations_controller.php

```diff
@@ -348,6 +349,207 @@ function export($type, $id)
+        if (!$course) {
+            $this->Session->setFlash(__('Error: The associated course does not exist or you do not have permission to view this course.', true));
+            $this->redirect('index');
+            return;
+        }
+        $course = $course['Course'];
+
+        $this->set('breadcrumb', $this->breadcrumb->push(array('course' => $course))
+             ->push(array('event' => $event))
+             ->push(__('Push Grades to Canvas', true)));
+
+        // if form is submitted, process it
+        if (isset($this->params['form']['submit'])) {
+
+            $exportReport = array();
+            $errorCount = 0;
+
+            $eventTemplateType = $this->Event->getEventTemplateTypeId($eventId);
+
+            $points_possible = 100;
+            if ($eventTemplateType == 1) {
+                $simpleEvaluation = $this->SimpleEvaluation->find('first', array(
+                    'conditions' => array('id' => $event['template_id']),
+                    'contain' => false,
+                ));
+                $points_possible = $simpleEvaluation['SimpleEvaluation']['point_per_member'];
+            } else if ($eventTemplateType == 2) {
+                $rubric = $this->Rubric->getRubricById($event['template_id']);
+                $points_possible = $rubric['Rubric']['total_marks'];
+            } else if ($eventTemplateType == 4) {
+                $mixeval = $this->Mixeval->getEvaluation($event['template_id']);
+                $points_possible = $mixeval['Mixeval']['total_marks'];
+            }
+
+            // Get Canvas course
+            if (!$course['canvas_id']) {
+                $this->Session->setFlash(__('Error: You need to first associate this course with a Canvas course in order to push grades. You can do so from the course Edit screen.', true));
+                $this->redirect('index');
+                return;
+            }
+            $canvasCourse = CanvasCourseComponent::getById($this, User::get('id'), $course['canvas_id']);
+            if (!$canvasCourse) {
+                $this->Session->setFlash(__('Error: Canvas course not found or you do not have access to it.', true));
+                $this->redirect('index');
+                return;
+            }
+
+            // Get Canvas assignment
+            $canvasAssignment = null;
+            if ($event['canvas_assignment_id']) {
+                $canvasAssignment = $canvasCourse->getAssignment($this, User::get('id'), $event['canvas_assignment_id']);
+            }
+
+            // Create assignment in Canvas if no assignment already associated or if assignment does not exist in Canvas
+            if (!$canvasAssignment) {
+                $assignmentArgs = array(
+                    'integration_id' => $eventId,
+                    'name' => $event['title'],
+                    'description' => $event['description'],
+                    'published' => $event['is_released'],
```

Comment:
If the instructor tried to push the grades after the iPeer event release date, the Canvas assignment will be created as "unpublished" and grades can't be updated.  Users will see an error message `user not authorized to perform that action`